File: 1768/solution.ts
function mergeAlternately(word1: string, word2: string): string {
  const res: Array<string> = [];
  let i = 0,
    j = 0;
  while (i < word1.length || j < word2.length) {
    if (i < word1.length) {
      res.push(word1[i++]);
    }
    if (j < word2.length) {
      res.push(word2[j++]);
    }
  }
  return res.join("");
}

File: 1768/solution.py
class Solution:
    def mergeAlternately(self, word1: str, word2: str) -> str:
        i, j = 0, 0
        res = []
        ONE, TWO = len(word1), len(word2)
        while i < ONE or j < TWO:
            if i < ONE:
                res.append(word1[i])
                i += 1
            if j < TWO:
                res.append(word2[j])
                j += 1
        return "".join(res)
